// @revolut/checkout/cjs/constants.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.LOCALES = [
    'en',
    'en-US',
    'nl',
    'fr',
    'de',
    'cs',
    'it',
    'lt',
    'pl',
    'pt',
    'es',
    'hu',
    'sk',
    'ja',
    'sv',
    'bg',
    'ro',
    'ru',
    'el',
    'hr',
    'tr',
];
exports.MODE = {
    PRODUCTION: 'prod',
    DEVELOPMENT: 'dev',
    SANDBOX: 'sandbox',
};
exports.URLS = (_a = {},
    _a[exports.MODE.SANDBOX] = {
        embed: 'https://sandbox-merchant.revolut.com/embed.js',
        version: 'https://sandbox-merchant.revolut.com/version.js',
        upsell: 'https://sandbox-merchant.revolut.com/upsell/embed.js',
    },
    _a[exports.MODE.PRODUCTION] = {
        embed: 'https://merchant.revolut.com/embed.js',
        version: 'https://merchant.revolut.com/version.js',
        upsell: 'https://merchant.revolut.com/upsell/embed.js',
    },
    _a[exports.MODE.DEVELOPMENT] = {
        embed: 'https://merchant.revolut.codes/embed.js',
        version: 'https://merchant.revolut.codes/version.js',
        upsell: 'https://merchant.revolut.codes/upsell/embed.js',
    },
    _a);
exports.REVOLUT_PAY_ORDER_ID_URL_PARAMETER = '_rp_oid';
exports.REVOLUT_PAY_SUCCESS_REDIRECT_URL_PARAMETER = '_rp_s';
exports.REVOLUT_PAY_FAILURE_REDIRECT_URL_PARAMETER = '_rp_fr';
